src/npf/npf_dataplane_session.c

```diff
@@ -113,13 +113,20 @@ static int dps_feature_nat_info(void *data, uint32_t *taddr, uint16_t *tport)
 	return npf_session_feature_nat_info(se, taddr, tport);
 }
 
+static void dps_feature_query(struct dp_session_info *info, struct session *s,
```

Comment:
Then your changes to this file probably would not be needed, other than having a way to map to/from an npf_session and a dp_dession - I suspect that already exists.

So it would result in the DPI code mapping to a dp_session, then using that to find its own feature block hanging off the dp_session, where your new "info" field would reside.